javascript/load-gen.ts

```diff
@@ -0,0 +1,406 @@
+      //  expand this to additional types of behavior
+      switch (result) {
+        case AsyncSetGetResult.SUCCESS:
+          successCount++;
+          loadGenContext.globalSuccessCount++;
+          break;
+        case AsyncSetGetResult.UNAVAILABLE:
+          loadGenContext.globalUnavailableCount++;
+          break;
+        case AsyncSetGetResult.DEADLINE_EXCEEDED:
+          loadGenContext.globalDeadlineExceededCount++;
+          break;
+        case AsyncSetGetResult.RESOURCE_EXHAUSTED:
+          loadGenContext.globalResourceExhaustedCount++;
+          break;
+        case AsyncSetGetResult.RST_STREAM:
+          loadGenContext.globalRstStreamCount++;
+          break;
+      }
+      if (loadGenContext.globalRequestCount % 1000 === 0) {
+        this.logger.info(`
+stats:
+       total: ${
+         loadGenContext.globalRequestCount
+       } (${BasicJavaScriptLoadGen.tps(
+          loadGenContext,
+          loadGenContext.globalRequestCount
+        )} tps)
+     success: ${
+       loadGenContext.globalSuccessCount
+     } (${BasicJavaScriptLoadGen.percentRequests(
+          loadGenContext,
+          loadGenContext.globalSuccessCount
+        )}%) (${BasicJavaScriptLoadGen.tps(
+          loadGenContext,
+          loadGenContext.globalSuccessCount
+        )} tps)
+ unavailable: ${
+   loadGenContext.globalUnavailableCount
+ } (${BasicJavaScriptLoadGen.percentRequests(
+          loadGenContext,
+          loadGenContext.globalUnavailableCount
+        )}%)
+deadline exceeded: ${
+          loadGenContext.globalDeadlineExceededCount
+        } (${BasicJavaScriptLoadGen.percentRequests(
+          loadGenContext,
+          loadGenContext.globalDeadlineExceededCount
+        )}%)
+resource exhausted: ${
+          loadGenContext.globalResourceExhaustedCount
+        } (${BasicJavaScriptLoadGen.percentRequests(
+          loadGenContext,
+          loadGenContext.globalResourceExhaustedCount
+        )}%)
+  rst stream: ${
+    loadGenContext.globalRstStreamCount
+  } (${BasicJavaScriptLoadGen.percentRequests(
+          loadGenContext,
+          loadGenContext.globalRstStreamCount
+        )}%)
+
+set latencies:      
+${BasicJavaScriptLoadGen.outputHistogramSummary(loadGenContext.setLatencies)}
+
+get latencies:
+${BasicJavaScriptLoadGen.outputHistogramSummary(loadGenContext.getLatencies)}
+`);
+      }
+    }
+    return successCount;
+  }
+
+  private async issueAsyncSetGet(
+    client: SimpleCacheClient,
+    loadGenContext: BasicJavasScriptLoadGenContext,
+    workerId: number,
+    requestId: number
+  ): Promise<AsyncSetGetResult> {
+    const cacheKey = `request${requestId}`;
+    try {
+      const setStartTime = process.hrtime();
+      await client.set(this.cacheName, cacheKey, this.cacheValue);
+      const setEndTime = process.hrtime(setStartTime);
+      const setDuration = (setEndTime[0] * 1e9 + setEndTime[1]) / 1e6;
```

Comment:
Nit: Does this end up being ms we end up with? Maybe hide the `1e9` and `1e6` math in a function make code bit more readable. ik gonna refactor this more so not big deal.

Reply:
totally agree, I thought I'd already pulled that out to a function. will do.

Reply:
done in d423998132f04e68fb33060da3821826b95fc37c